src/otx/core/model/entity/base.py

```diff
@@ -106,3 +109,62 @@ def map_class_names(src_classes: list[str], dst_classes: list[str]) -> list[int]
             else:
                 src2dst.append(-1)
         return src2dst
+
+
+class OVModel(OTXModel, Generic[T_OTXBatchDataEntity, T_OTXBatchPredEntity]):
+    """Base class for the OpenVINO model."""
+
+    def __init__(self, config: DictConfig) -> None:
+        self.model_name = config.pop("model_name")
+        self.model_type = config.pop("model_type")
+        self.async_inference = config.pop("async_inference", False)
+        self.num_requests = config.pop("max_num_requests", get_default_async_reqs_num())
+        self.use_tp_mode = config.pop("use_tp_mode", False)
+        self.config = config
+        super().__init__()
+
+    def _create_model(self) -> nn.Module:
+        """Create a OV model with help of Model API."""
+        from openvino.model_api.adapters import OpenvinoAdapter, create_core, get_user_config
+        from openvino.model_api.models import Model
+
+        plugin_config = get_user_config("AUTO", str(self.num_requests), "AUTO")
+        if self.use_tp_mode:
+            plugin_config["PERFORMANCE_HINT"] = "THROUGHPUT"
+
+        model_adapter = OpenvinoAdapter(
+            create_core(),
+            self.model_name,
+            max_num_requests=self.num_requests,
+            plugin_config=plugin_config,
+        )
+
+        return Model.create_model(model_adapter, model_type=self.model_type)
+
+    def _customize_inputs(self, entity: T_OTXBatchDataEntity) -> dict[str, Any]:
+        # restore original numpy image
+        images = [np.transpose(im.numpy(), (1, 2, 0)) for im in entity.images]
+        return {"inputs": images}
+
+    def forward(
+        self,
+        inputs: T_OTXBatchDataEntity,
+    ) -> T_OTXBatchPredEntity | OTXBatchLossEntity:
+        """Model forward function."""
+
+        def _callback(result: NamedTuple, user_data: list[NamedTuple]) -> None:
+            user_data.append(result)
+
+        numpy_inputs = self._customize_inputs(inputs)["inputs"]
+        if self.async_inference:
+            outputs: list[Any] = []
+            self.model.set_callback(_callback)
+            for im in numpy_inputs:
+                if not self.model.is_ready():
+                    self.model.await_any()
+                self.model.infer_async(im, user_data=outputs)
+            self.model.await_all()
+        else:
+            outputs = [self.model(im) for im in numpy_inputs]
+
+        return self._customize_outputs(outputs, inputs)
```

Comment:
Is it still synchronous for each batch? I'd rather to make `self.async_inference=True` as the default (batch-wise synchronized inference) for `forward()`. Then, create `def forward_async()` which is truly asynchronous, so that return a future ofmodel output. If it is hard to develop `def forward_async()` this time. Please do it later.

Reply:
why we should create forward_async()? It works OK in forward() as well. self.async_inference depends on parameter in recipe.yaml. By default it is True for all tasks.

Reply:
You are doing this
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/f990f247-e039-4568-8dbc-e0881ceb448c)
But I was asking you change it as a default (implementation) of `forward()` and create this one
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/765e908c-d8ec-43d4-b103-4eda6f645171)

Reply:
Let's discuss it later, we now need to focus on export